sky/packages/sky/lib/src/widgets/event_recorder.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+
+enum EventRecorderMode {
+  stop,
+  record
+}
+
+typedef void EventsReady(List<PointerInputEvent> events);
+
+class EventRecorder extends StatefulComponent {
+  EventRecorder({
+    Key key,
+    this.child,
+    this.mode: EventRecorderMode.stop,
+    this.onEventsReady
+  });
+
+  final Widget child;
+  final EventRecorderMode mode;
+  final EventsReady onEventsReady;
+
+  _EventRecorderState createState() => new _EventRecorderState();
+}
+
+class _EventRecorderState extends State<EventRecorder> {
+
+  EventRecorderMode _mode;
+  List<PointerInputEvent> _events = new List<PointerInputEvent>();
+
+  void initState() {
+    super.initState();
+    _mode = config.mode;
+  }
+
+  void didUpdateConfig(EventRecorder oldConfig) {
+    if (_mode == EventRecorderMode.record &&
+        config.mode == EventRecorderMode.stop) {
+      config.onEventsReady(_events);
+      _events.clear();
+    }
+    _mode = config.mode;
```

Comment:
There's no need to store the _mode on `this`.  You can check the old value of mode from `oldConfig.mode`.